File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/search/SearchUIState.kt
package dev.subfly.yabacore.state.search

import androidx.compose.runtime.Immutable
import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType

@Immutable
data class SearchUIState(
    val query: String = "",
    val bookmarks: List<BookmarkUiModel> = emptyList(),
    val selectedFolderIds: Set<String> = emptySet(),
    val selectedTagIds: Set<String> = emptySet(),
    val bookmarkAppearance: BookmarkAppearance = BookmarkAppearance.LIST,
    val sortType: SortType = SortType.EDITED_AT,
    val sortOrder: SortOrderType = SortOrderType.DESCENDING,
    val isLoading: Boolean = false,
)

File: Darwin/YABACore/State/Search/SearchUIState.swift
//
//  SearchUIState.swift
//  YABACore
//

import Foundation

public struct SearchUIState: Sendable {
    public var query: String
    public var enabledFolderFilterIds: Set<String>
    public var enabledTagFilterIds: Set<String>
    public var sortType: SortType
    public var sortOrder: SortOrderType
    public var bookmarkAppearance: BookmarkAppearance
    public var cardImageSizing: CardImageSizing?

    public init(
        query: String = "",
        enabledFolderFilterIds: Set<String> = [],
        enabledTagFilterIds: Set<String> = [],
        sortType: SortType = .editedAt,
        sortOrder: SortOrderType = .descending,
        bookmarkAppearance: BookmarkAppearance = .list,
        cardImageSizing: CardImageSizing? = .small
    ) {
        self.query = query
        self.enabledFolderFilterIds = enabledFolderFilterIds
        self.enabledTagFilterIds = enabledTagFilterIds
        self.sortType = sortType
        self.sortOrder = sortOrder
        self.bookmarkAppearance = bookmarkAppearance
        self.cardImageSizing = cardImageSizing
    }
}
